tests/__init__.py

```diff
@@ -144,3 +144,13 @@ class Module:
 @dataclass
 class Package:
     modules: List[Module] = MISSING
+
+
+@dataclass
+class SubscriptedList:
+    list: List[int] = field(default_factory=lambda: [1, 2])
+
+
+@dataclass
+class SubscriptedDict:
+    dict: Dict[str, int] = field(default_factory=lambda: {"foo": 4})
```

Comment:
Curious, cab you check what happens to dataclasses that extends `Dict[str, int]` for example?

Reply:
Pickling this class works just fine because the ref_type stored is `DictSubclass`
```python
@dataclass
class DictSubclass(Dict[str, int]):
    pass
```

It should be `ref_type=Dict[str, int]` since obj_type is `DictSubclass` imo.

Reply:
ref_type for structured configs is determined by the annotation. I think that if there is no annotation it defaults to the dataclass itself.

What are you actually seeing?

Reply:
`DictClass(Dict[str, int])` ref_type is DictSubclass instead of `Dict[str, int]`

```python
@dataclass
class DictSubclass(Dict[str, int]):
    pass
from omegaconf._utils import get_ref_type
x = OmegaConf.structured(DictSubclass())
print(x._metadata.ref_type)
print(get_ref_type(x))

<class '__main__.DictSubclass'>
typing.Union[__main__.DictSubclass, NoneType]
``

Reply:
This seems correct to me.

Reply:
Which is exaclty what I was expecting.